internal/service/wallet.go

```diff
@@ -447,28 +465,134 @@ func (s *walletService) GetWalletBalance(ctx context.Context, walletID string) (
 			continue
 		}
 
-		if usageResp.Amount > 0 {
-			currentPeriodUsage = currentPeriodUsage.Add(decimal.NewFromFloat(usageResp.Amount))
+		// STEP 4: Get billable amount from billing service
+		// Note: This returns what will be invoiced, which might be different
+		// from what we need to deduct from wallet
+		_, billableAmount, err := billingService.CalculateUsageCharges(
+			ctx,
+			sub.Subscription,
+			usageResp,
+			sub.Subscription.CurrentPeriodStart,
+			sub.Subscription.CurrentPeriodEnd,
+		)
+		if err != nil {
+			s.Logger.Errorw("failed to calculate billable charges",
+				"wallet_id", walletID,
+				"subscription_id", sub.ID,
+				"raw_usage_amount", usageResp.Amount,
+				"error", err,
+			)
+			continue
 		}
+
+		s.Logger.Debugw("billing service calculation details",
+			"wallet_id", walletID,
+			"subscription_id", sub.ID,
+			"raw_usage_amount", usageResp.Amount,
+			"billable_amount", billableAmount,
+			"conversion_rate", w.ConversionRate,
+			"currency", w.Currency,
+		)
+
+		// STEP 5: Calculate what to deduct from wallet
+		// We need this because billableAmount might be zero for two reasons:
+		// 1. Usage is within entitlement (don't deduct)
+		// 2. Usage will be billed later (should deduct)
+		actualUsageAmount := billableAmount
+		if billableAmount.IsZero() && usageResp.Amount > 0 {
```

Comment:
remove this codeblock and test and if we can think of any issue then only lets add this